### DISCUSSION TITLE
Database changes for v1.6.5

### DISCUSSION THREAD

**Discussion by n-stein** [moneymanagerex/moneymanagerex#6058] [General]
#5439 requires database changes to add new tables for tagging transactions. Are there any other database changes we should make for v19 for the 1.6.5 release?

**Comment by vomikan** [comment]
Colours

**Reply by n-stein** [reply]
Which table/field? Is there an issue for reference?

**Reply by vomikan** [reply]
Its followupid in checkingaccount table.
Plus the same for recurring transactions.

**Reply by n-stein** [reply]
Sorry @vomikan, I don't understand. What needs to be changed about CHECKINGACCOUNT_V1.FOLLOWUPID or BILLSDEPOSITS_V1.FOLLOWUPID?

**Comment by vomikan** [comment]
> What needs to be changed about CHECKINGACCOUNT_V1.FOLLOWUPID or BILLSDEPOSITS_V1.FOLLOWUPID?

The name of the field does not correspond to the purpose. We need to add the color field to both of these tables and transfer the data there from the followupid field.
It may even be better to store in the color field not the color ID, but its value, for example, #FFEE00

**Comment by n-stein** [comment]
> It may even be better to store in the color field not the color ID, but its value, for example, #FFEE00

I like this, but maybe we keep COLOR as integer and store either the user color ID (-1 through 7) or the packed RGB value. This allows me to do this:

![mmex_bqssL9WmOo](https://github.com/moneymanagerex/moneymanagerex/assets/16599929/6b4d2b95-2f0a-4fa8-8a45-85757dd4da06)

This has the added benefit of making copy from FOLLOWUPID -> COLOR simple.

**Comment by n-stein** [comment]
PR moneymanagerex/database#52

**Comment by DevAnyKR** [comment]
transaction table - date format.
change to datetime format
